twitter4j-stream/src/main/java/twitter4j/TwitterStreamImpl.java

```diff
@@ -78,20 +78,18 @@ public StatusStream getStream() throws TwitterException {
      * {@inheritDoc}
      */
     @Override
-    public void geohose(final int count, final String partitions) {
+    public void geohose(final int count, final String partition) {
         ensureAuthorizationEnabled();
         ensureStatusStreamListenerIsSet();
         startHandler(new TwitterStreamConsumer(Mode.status) {
             @Override
             public StatusStream getStream() throws TwitterException {
                 ensureAuthorizationEnabled();
                 final List<HttpParameter> params = new ArrayList<HttpParameter>();
-                params.add(new HttpParameter("include_fields", "public_location"));
-                params.add(new HttpParameter("count", String.valueOf(count)));
-                if (null != partitions) {
-                    params.add(new HttpParameter("partitions", partitions));
+                if (null != partition) {
+                    params.add(new HttpParameter("partition", partition));
                 }
-                return getCountStream("statuses/firehose.json", params.toArray(new HttpParameter[params.size()]));
+                return getCountStream("stream/firehose/accounts/NielsenTV/publishers/twitter/prod.json", params.toArray(new HttpParameter[params.size()]));
```

Comment:
Also since parameters of Twitter geohose and gnip are different, for example in terms of the backfill options, instead of modyfying geohose method we could write our own one for Gnip.